var/spack/repos/builtin/packages/amd-aocl/package.py

```diff
@@ -38,21 +39,32 @@ class AmdAocl(BundlePackage):
         depends_on("amdblis threads=openmp")
         depends_on("amdfftw +openmp")
         depends_on("amdlibflame threads=openmp")
+        depends_on("aocl-sparse +openmp")
+        depends_on("aocl-da +openmp")
+        depends_on("aocl-compression +openmp")
 
     with when("~openmp"):
         depends_on("amdblis threads=none")
         depends_on("amdfftw ~openmp")
         depends_on("amdlibflame threads=none")
+        depends_on("aocl-sparse ~openmp")
+        depends_on("aocl-da ~openmp")
+        depends_on("aocl-compression ~openmp")
 
-    for vers in ["2.2", "3.0", "3.1", "3.2", "4.0", "4.1", "4.2"]:
+    for vers in ["2.2", "3.0", "3.1", "3.2", "4.0", "4.1", "4.2", "5.0"]:
         with when(f"@={vers}"):
             depends_on(f"amdblis@={vers}")
             depends_on(f"amdfftw@={vers}")
             depends_on(f"amdlibflame@={vers}")
+            depends_on("amdlibflame ^[virtuals=blas] amdblis")
```

Comment:
Where these changes validated? It turns out we have a bug in an audit that let these pass, but we never allowed `^` in depends_on directives, and these `depends_on` are treated internally as:
```python
depends_on("amdlibflame")
```
due to `deps=False` in:

https://github.com/spack/spack/blob/fc4316cafaccc591f34d6dbbeabd0cad0427b340/lib/spack/spack/directives.py#L304